internal/declcfg/declcfg.go

```diff
@@ -86,6 +88,11 @@ func (m *Meta) UnmarshalJSON(blob []byte) error {
 	}
 	m.Schema = t.Schema
 	m.Package = t.Package
+	m.Name = t.Name
+	// Package name is encoded as "name" in the package schema.
+	if m.Schema == schemaPackage {
+		m.Package = t.Name
+	}
```

Comment:
Not sure I'm convinced this assignement is really necessary. `olm.package` blobs don't have a package field, so IMO it would be fine for this field to just be an empty string for packages.